pkg/rest/config_upload.go

```diff
@@ -262,17 +266,41 @@ func isPaginatedResponse(jsonResponse map[string]interface{}) (paginated bool, p
 	return false, ""
 }
 
-func translateGenericValues(inputValues []interface{}) []api.Value {
+func translateGenericValues(inputValues []interface{}, configType string) ([]api.Value, error) {
+
 	numValues := len(inputValues)
 	values := make([]api.Value, numValues, numValues)
+
 	for i := 0; i < numValues; i++ {
 		input := inputValues[i].(map[string]interface{})
+
+		if input["id"] == nil {
+			return values, fmt.Errorf("config of type %s was invalid: No id", configType)
+		}
+
+		// repair invalid configs - but log them
+		if input["name"] == nil {
+			jsonStr, err := json.Marshal(input)
+			if err != nil {
+				util.Log.Warn("Could not marshal %s", configType)
```

Comment:
This log is not particularly user friendly IMHO; could we log similarly to the log on line 289? 
```suggestion
				util.Log.Warn("Config of type %s was invalid. Ignoring it!", configType)
```